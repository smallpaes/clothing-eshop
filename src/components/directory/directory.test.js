import { shallow } from 'enzyme';

import Directory from './directory.component';

describe('Directory component', () => {
  const mockSections = [
    {
      id: 1,
      title: 'title', 
      imageUrl: 'www.test.com', 
      size: 'large'
    },
    {
      id: 2,
      title: 'title', 
      imageUrl: 'www.test.com', 
      size: 'large'
    },
    {
      id: 3,
      title: 'title', 
      imageUrl: 'www.test.com', 
      size: 'large'
    },
    {
      id: 4,
      title: 'title', 
      imageUrl: 'www.test.com', 
      size: 'large'
    }
  ]

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Directory sections={mockSections} />)
  });

  it('Should render PreviewContainer component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the correct amount of component inside DirectoryContainer', () => {
    expect(wrapper.find('DirectoryContainer').children()).toHaveLength(mockSections.length);
  });
});