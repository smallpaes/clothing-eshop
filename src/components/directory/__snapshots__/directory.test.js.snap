// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component Should render PreviewContainer component 1`] = `
<DirectoryContainer>
  <withRouter(MenuItem)
    imageUrl="www.test.com"
    key="1"
    size="large"
    title="title"
  />
  <withRouter(MenuItem)
    imageUrl="www.test.com"
    key="2"
    size="large"
    title="title"
  />
  <withRouter(MenuItem)
    imageUrl="www.test.com"
    key="3"
    size="large"
    title="title"
  />
  <withRouter(MenuItem)
    imageUrl="www.test.com"
    key="4"
    size="large"
    title="title"
  />
</DirectoryContainer>
`;
